opaque_keys/edx/keys.py

```diff
@@ -89,6 +89,36 @@ def make_asset_key(self, asset_type: str, path: str) -> AssetKey:  # pragma: no
         raise NotImplementedError()
 
 
+class LibCollectionKey(LearningContextKey):
+    """
+    An :class:`opaque_keys.OpaqueKey` identifying a particular Library Collection object.
+    """
+    __slots__ = ()
+
+    @property
+    @abstractmethod
+    def org(self) -> str | None:  # pragma: no cover
+        """
+        The organization that this collection belongs to.
+        """
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def lib(self) -> str | None:  # pragma: no cover
+        """
+        The lib for this collection.
+        """
+        raise NotImplementedError()
+
+    @property
+    def context_key(self) -> LearningContextKey:  # pragma: no cover
+        """
+        Get the learning context key (LearningContextKey) for this XBlock usage.
+        """
+        raise NotImplementedError()
```

Comment:
It's not clear to me that a library is the "collection" of the context--I think that's mixing metaphors. Since we're making the assumption that a collection always belongs to a library, I would just be concrete and use the `lib_key` field that's already defined on your locator below. And, going along with my other request, I would spell it out as `library_key`.
```suggestion
    @property
    def library_key(self) -> LibraryLocatorV2:  # pragma: no cover
        """
        Get the key of the library that this collection belongs to.
        """
        raise NotImplementedError()
```